cpp/src/io/avro/avro.cpp

```diff
@@ -57,14 +90,16 @@ bool container::parse(file_metadata *md, size_t max_num_rows, size_t first_row)
       }
     }
   }
-  for (int i = 0; i < 16; i++) { (reinterpret_cast<uint8_t *>(&md->sync_marker[0]))[i] = getb(); }
+  md->sync_marker[0] = get_raw<uint64_t>();
+  md->sync_marker[1] = get_raw<uint64_t>();
+
   md->metadata_size  = m_cur - m_base;
   md->skip_rows      = 0;
   max_block_size     = 0;
   total_object_count = 0;
   while (m_cur + 18 < m_end && total_object_count < max_num_rows) {
```

Comment:
Not part of your PR, but 18 is a truly magic number to ever spot in the wild.  Is it possible this is supposed to be a 16?

Reply:
:D
I think you're right, since we read two 64bit values in the loop here (at least). I'm not 100% sure since I haven't looked into the format (yet).